src/test/java/org/minions/devfund/carlosorellana/WhackAMoleTest.java

```diff
@@ -0,0 +1,327 @@
+    }
+
+    /**
+     * Validate score when Two moles was found and One mole is incorrect.
+     */
+    @Test
+    public void validateGetScoreWhenTwoCorrectMolesFoundAndOneIncorrect() {
+        final int xPosMole1 = 7;
+        final int yPosMole1 = 9;
+
+        final int xPosMole2 = 5;
+        final int yPosMole2 = 5;
+
+        final int xPosMole3 = 9;
+        final int yPosMole3 = 1;
+
+        final int xPosMole3Failed = 1;
+        final int yPosMole3Failed = 9;
+
+        final int expectedResult = 2;
+        whackAMole.place(xPosMole1, yPosMole1);
+        whackAMole.place(xPosMole2, yPosMole2);
+        whackAMole.place(xPosMole3, yPosMole3);
+
+        whackAMole.whack(xPosMole1, yPosMole1);
+        whackAMole.whack(xPosMole2, yPosMole2);
+        whackAMole.whack(xPosMole3Failed, yPosMole3Failed);
+
+        assertEquals(expectedResult, whackAMole.getScore());
+    }
+
+    /**
+     * Validate the Moles Left found.
+     */
+    @Test
+    public void validateGetScoreWhenOneCorrectMolesFoundAndOneIncorrectResult() {
+        final int xPosMole1 = 7;
+        final int yPosMole1 = 9;
+
+        final int xPosMole2 = 9;
+        final int yPosMole2 = 1;
+
+        final int xPosMole2Failed = 1;
+        final int yPosMole2Failed = 9;
+
+        final int expectedResult = 1;
+        whackAMole.place(xPosMole1, yPosMole1);
+        whackAMole.place(xPosMole2, yPosMole2);
+
+        whackAMole.whack(xPosMole1, yPosMole1);
+        whackAMole.whack(xPosMole2Failed, yPosMole2Failed);
+
+        assertEquals(expectedResult, whackAMole.getMolesLeft());
+    }
+
+    /**
+     * Validate the message if the number of attempts is greater than 50.
+     */
+    @Test
+    public void validateGetAttemptsMessage() {
+        String  messageAttempts = "The number of attempts exceeds the allowed";
+
+        final int xPosMole1 = 7;
+        final int yPosMole1 = 9;
+
+        int attempts = NUM_ATTEMPTS;
+
+        while (attempts > 0) {
+            whackAMole.place(xPosMole1, yPosMole1);
+            whackAMole.whack(xPosMole1, yPosMole1);
+            attempts--;
+        }
+
+        whackAMole.whack(xPosMole1, yPosMole1);
+
+        assertEquals(messageAttempts, whackAMole.getMessageAttempts());
+    }
+
+    /**
+     * validate Place Result is True when the positions are correct.
+     */
+    @Test
+    public void validatePlaceResultIsTrueWhenPositionsAreCorrect() {
+        final int xPos = 10;
+        final int yPos = 10;
+
+        assertTrue("The Position send to grid are incorrect", whackAMole.place(xPos, yPos));
+    }
+
+    /**
+     * validate Place Result is False when the mole was already placed.
+     */
+    @Test
+    public void validatePlaceResultIsFalseWhenMolesWasAlreadyPlaced() {
+        final int xPos = 5;
+        final int yPos = 5;
+        whackAMole.place(xPos, yPos);
+
+        assertFalse("The Position send to grid are correct", whackAMole.place(xPos, yPos));
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "The Position send to grid are correct" 5 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)